keygen/openssl.conf

```diff
@@ -1,4 +1,41 @@
 [req]
 distinguished_name = req_distinguished_name
+x509_extensions = v3_ca	# The extentions to add to the self signed cert
 
 [req_distinguished_name]
+
+[ v3_ca ]
+
+# Extensions for a typical CA
+
+
+# PKIX recommendation.
+
+subjectKeyIdentifier=hash
+
+authorityKeyIdentifier=keyid:always,issuer
+
+# This is what PKIX recommends but some broken software chokes on critical
+# extensions.
+#basicConstraints = critical,CA:true
+# So we do this instead.
+basicConstraints = CA:true
+
+# Key usage: this is typical for a CA certificate. However since it will
+# prevent it being used as an test self-signed certificate it is best
+# left out by default.
+# keyUsage = cRLSign, keyCertSign
```

Comment:
Let's distinguish comments and commented out parameters. For instance, comments could start with two hashes. Or the commented out parameters should not have a space after hash.